### ISSUE TITLE
Roadmap for Microsoft365DSC ?

### ISSUE THREAD

**Issue by salbeck-sit** [microsoft/Microsoft365DSC#5140]
I have received notice that Microsoft will discontinue PSDSC sometime in '27 and since we have a number of tenants that we want to configure and maintain using Microsoft365DSC, the question has popped up: What will happen to Microsoft365DSC?

It's interesting first to look to CloudConfigurationManager, secondly to DSCv3. The former seems an easier way to transition but I may be missing out since I haven't spent much time with DSCv3 (which seems to promise to support PSDSC-based resources).

Anybody want to shine a light on this issue?

**Comment by WavingReader** [comment]
If there's something to it, I'm interested too. We want to use M365DSC on a larger scale.

**Comment by Borgquite** [comment]
@salbeck-sit Is this the discontinuation you're talking about - or is there another one I haven't seen?

> Azure Automation State Configuration will be retired on September 30, 2027, please transition to [Azure Machine Configuration](https://learn.microsoft.com/en-us/azure/governance/machine-configuration/overview) by that date.

https://learn.microsoft.com/en-us/azure/automation/automation-dsc-overview
https://azure.microsoft.com/en-us/updates/v2/Planned-Service-Retirement-Azure-Automation-State-Configuration-16-September-2027

**Comment by salbeck-sit** [comment]
@Borgquite  Not too sure what it's all about. I don't think it's related only to Automation.

I received word of this from a colleague that heard it from a partner. It has been debated here some time ago but I can't find it. The last article you linked to says that it isn't related to PSDSC. See https://learn.microsoft.com/en-us/powershell/dsc/overview?view=dsc-3.0 (quote: 'DSCv3 is a platform tool that abstracts the concerns for defining and invoking resources. Higher order tools, like Azure Machine Configuration, Azure Automanaged VM, and WinGet are early partners for DSCv3 as orchestration agents.')

I'm aware of the shift to DSC v2.0 outlined in https://microsoft365dsc.com/user-guide/get-started/powershell7-support/ but that's not a concern right now.

**Comment by Borgquite** [comment]
Yeah - haven't personally seen or heard anything about deprecation in PowerShell beyond the above but would be helpful if you can find a source.

The obvious thing that could be deprecated is Windows PowerShell 5.1 which would also involve deprecation of the LCM, but haven't seen a solid announcement.

**Comment by Borgquite** [comment]
There’s some related discussion on both the switch to PS Core 7+ and how PS 5.1 and script-based resources like Microsoft365DSC are supported in DSCv3 in #5433

**Comment by FabienTschanz** [comment]
I'll be closing this in favor of other issues that have more discussion context in it.
